core/dbt/source_config.py

```diff
@@ -50,9 +50,16 @@ def __init__(self, active_project, own_project, fqn, node_type):
     def _merge(self, *configs):
         merged_config = {}
         for config in configs:
+            # Do not attempt to deep merge clobber fields
```

Comment:
@beckjake I preserved the `copy` here since we pop elements out of the dict in the comprehension below. Can you give this another quick look?

Reply:
Yes, I think that's the correct behavior